UI Improvements with Tailwind CSS Integration

garvitgabgab opened the pull request:
Refactored the existing UI to use Tailwind CSS for improved styling consistency and maintainability.
Made adjustments to layout, spacing, and alignment to enhance the overall user experience.

![Screenshot 2024-09-23 152805](https://github.com/user-attachments/assets/471d8572-d104-415a-b1a6-0b8ceb65970f)
![Screenshot 2024-09-23 152825](https://github.com/user-attachments/assets/af54a110-2ebe-4854-af40-091a69745cac)

Ezio-Sarthak commented:
Thanks for working on this @garvitgabgab ! Before jumping into the review here are few things I wish to convey:
- It's a fix that I was supposed to do while creating the repo, but could you please add a gitignore file with `node_modules` folder added. Right now I see 5000+ file changes.
- Let's make PR title short and keep other things in the PR description. It's a good practice that way.
- Please try to add commit description to all commits.
- I see lot of code changes in a single commit. Can we split these into small meaningful commits?

Kindly go through these comments and let me know 👍

garvitgabgab commented:
Hey @Ezio-Sarthak , thanks for your feedback! Here’s my response to each of your points:

-      I’ve added the .gitignore file with the node_modules folder excluded. The 5000+ file changes were likely due to my oversight in not committing frequently. I realize now that the golden rule is to commit after every meaningful change.
- I shortened the PR title as suggested and moved the details into the PR description for clarity and to follow best practices.
- Regarding the suggestion to split code changes into smaller, meaningful commits—I've already made some major UI changes and didn't made commit during that but now I made a few minor changes and tried adding meaningful commits which were split across a few commits. Moving forward, I will focus on making sure that each commit covers a distinct, logical change, to maintain better clarity.


If there’s anything else I can improve on, please let me know! Looking forward to your review.

Ezio-Sarthak commented:
Hi @garvitgabgab apologies for the delayed response! Kindly rebase this branch against the latest main as it is showing some merge conflicts. I've mentioned the steps for rebasing in the discussions channel 👍
